kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -460,6 +503,32 @@ function isReplicaCountEqualToMinReplicas {
   fi
 }
 
+#
+# Function to check if provided replica count is in the allowed range
+# $1 - Domain resource in json format
+# $2 - Name of the cluster
+# $3 - Replica count
+# $4 - Returns "true" or "false" indicating if replica count is in
+#      the allowed range
+# $5 - Returns allowed range for replica count for the given cluster
+#
+function isReplicasInAllowedRange {
+  local domainJson=$1
+  local clusterName=$2
+  local replicas=$3
+  local __result=$4
+  local __range=$5
+
+  eval $__result=true
+  getMinReplicas "${domainJson}" "${clusterName}" minReplicas
+  getMaxReplicas "${domainJson}" "${clusterName}" maxReplicas
+  range="${minReplicas} to ${maxReplicas}"
+  if [ ${replicas} -lt ${minReplicas} ] || [ ${replicas} -gt ${maxReplicas} ]; then
+    eval $__result=false
+    eval $__range="'${range}'"
+  fi
```

Comment:
(1) It looks like range is supposed to be a local var.  Make it local and give it a unique name that doesn't overlap with any global.
(2) It looks like there's a missing 'eval' for $__range if we fall into the if-then stanza